src/tactic/linarith/preprocessing.lean

```diff
@@ -258,22 +258,49 @@ do s ← ls.mfoldr (λ h s', infer_type h >>= find_squares s') mk_rb_set,
    products ← make_comp_with_zero.globalize.transform products.reduce_option,
    return $ new_es ++ ls ++ products }
 
+/--
+`remove_ne_aux` case splits on any proof `h : a ≠ b` in the input, turning it into `a < b ∨ a > b`.
+This produces `2^n` branches when there are `n` such hypotheses in the input.
+-/
+meta def remove_ne_aux : list expr → tactic (list branch) :=
+λ hs,
+(do e ← hs.mfind (λ e : expr, do e ← infer_type e, guard $ e.is_ne.is_some),
+    [(_, ng1), (_, ng2)] ← to_expr ``(or.elim (lt_or_gt_of_ne %%e)) >>= apply,
+     let do_goal : expr → tactic (list branch) := λ g,
+       do set_goals [g],
+          h ← intro1,
+          ls ← remove_ne_aux $ hs.remove_all [e],
+          return $ ls.map (λ b : branch, (b.1, h::b.2)) in
+      (++) <$> do_goal ng1 <*> do_goal ng2)
```

Comment:
Indentation is a bit odd here, I would expect the `(++)` line to line up with the `let`, and that line to line up with the one before it.